docs/howto.rst

```diff
@@ -1,23 +1,141 @@
 How-to...
 =========
 
-Ensure tasks don't run concurrently and run in order
-----------------------------------------------------
+Ensure jobs don't run concurrently and run in order
+---------------------------------------------------
 
-Launch a task in the future
+Let's imagine we have a task like this::
+
+    @task_manager.task(queue="queue")
+    def write_alphabet(floor):
+        time.sleep(random.random() * 5)
+        with open("/tmp/alphabet.txt", "a") as f:
+            f.write(floor)
+
+We write the letter we recieve to a file after wating for a
+random time (this is a simplified version of a real
+world problem where tasks take a unforseeable amount of time
+and share resources like a database).
+
+We call it::
+
+    write_alphabet.defer("a")
+    write_alphabet.defer("b")
+    write_alphabet.defer("c")
+    write_alphabet.defer("d")
+
+We could expect the following to be written in `pyramid.txt`::
+
+    a
+    b
+    c
+    d
+
+And what we find is pretty much like this::
+
+    d
+    a
+    c
+    b
+
+Jobs were taken from the queue in order, but if we have several
+workers, they have been launched in parallel and given their duration
+is random, the final result pretty much is too.
+
+We can solve this problem by using locks::
+
+    job_descritption = write_alphabet.configure(lock="/tmp/alphabet.txt")
```

Comment:
using a path as lock parameter could be misleading no ? (even if it's the name of the resource, the task might have multiple resources used)

Reply:
Not sure what you mean... What do you suggest ?

Note: the whole task makes no sense in a distributed context, the file will be split between all the machines that run the workers.

Reply:
ok, keep this :)